### PR TITLE
✨ — Add "bouncers reset", "bouncers del" and "bouncers [--status] review" commands to sympa.pl

### LINKED ISSUE #1058: Add "bouncers reset", "bouncers del" and "bouncers [--status] review" commands to sympa.pl
Due to recent gmail problems, a lot of gmail.com subscribers are in error status in a lot of our lists.

Having a command line tool to cancel errors (and unsubscribe users in error status, since the option exists in the web UI) would be useful.

Note: I intend to code it myself next week, I’m creating this issue to think about it.

Expected Behavior
-----------------
Being able to cancel errors on unsubscribe users in error status from command line with `sympa.pl`.

Current Behavior
----------------
Such option does not exist.

### PR DESCRIPTION
**Pull request by ldidry**
Fix #1058 

I’m not sure about the `--show_status_of_members`: I print differently users with `bounce` attribute, users with `suspend` attribute and all the others. Is `suspend` attribute a sign of an error? Should I just print the `ok` and `bounce` users?

What are your thoughts about it?

### CODE REVIEW

**Comment by ikedas** [comment]
* The word "subscribe"/"unsubscribe" has been used for the actions by the user to add/remove themselves.  The word "add"/"del" has been used for the actions by administrators.
* I feel the options are way too long.

Anyway, how about submit a issue where we may discuss about your PR(s)?  I think the discussion may take place on that issue page, and each time when you come to any conclusion, you can submit a PR or push addition to exisiting PR.

**Comment by ldidry** [comment]
> The word "subscribe"/"unsubscribe" has been used for the actions by the user to add/remove themselves. The word "add"/"del" has been used for the actions by administrators.

No pb to change that

> I feel the options are way too long.

Indeed but I don’t see tha as a problem: sympa.pl has too many options to remember all of them, I think admins check the help to look at the available options. Don’t you think?

> Anyway, how about submit a issue where we may discuss about your PR(s)?

Well… why not using issue #1058?

**Comment by ikedas** [comment]
> > Anyway, how about submit a issue where we may discuss about your PR(s)?
> 
> Well… why not using issue #1058?

Oh yes, no problem.

File: src/sbin/sympa.pl.in [hunk 1]
```diff
@@ -179,6 +181,14 @@ if ($options{'add'}) {
...
 } elsif ($options{'version'}) {
     Sympa::CLI->run({}, 'version');
+} elsif ($options{'review'}) {
+    Sympa::CLI->run(\%options, 'review', $options{review});
+} elsif ($options{'reset_members_bounces'}) {
+    Sympa::CLI->run({}, 'bouncers', 'reset',
+        $options{'reset_members_bounces'});
+} elsif ($options{'del_bounced_members'}) {
+    Sympa::CLI->run({}, 'bouncers', 'del',
+        $options{'del_bounced_members'});
```
Comment (ikedas) [inline src/sbin/sympa.pl.in:191]: I personally think that the new features need not to support the old-style command line format.

Comment (ldidry) [reply src/sbin/sympa.pl.in:191]: Ok with that (die, old syntax, die! :sweat_smile: ). I’ll update the PR.

File: src/lib/Sympa/Request/Handler/del.pm [hunk 2]
```diff
@@ -126,7 +126,8 @@ sub _report_member {
                 $who);
         }
     }
-    $self->add_stash($request, 'notice', 'removed', {'email' => $who});
+    $self->add_stash($request, 'notice', 'removed',
+        {'email' => $who, 'listname' => $list->get_id});
```
Comment (ldidry) [inline src/lib/Sympa/Request/Handler/del.pm:130]: Just for information: `listname` is used in `default/mail_tt2/report.tt2` but was not defined, so I added it (I noticed it while using the new `bouncers del _list_` feature).

**Comment by 0137248506** [comment]
> [#1058'i ](https://github.com/sympa-community/sympa/issues/1058)düzelt[](https://github.com/sympa-community/sympa/issues/1058)
> 
> Şunlardan emin değilim : Özniteliğe sahip kullanıcıları, özniteliğe sahip kullanıcıları ve diğerlerini `--show_status_of_members`farklı şekilde yazdırıyorum . Özellik bir hatanın işareti midir ? Sadece ve kullanıcılarını yazdırmalı mıyım ?`bounce``suspend``suspend``ok``bounce`
> 
> Bu konuda düşünceleriniz neler?

Mwlakw zcxkwd krowz

**Comment by 0137248506** [comment]
Uploading e77863d8197b3adefa4a5d070b7ada4c.mp4…

**Pull request merged**